Sources/OktaIdxAuth/Internal/Requests/Authenticate.swift

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Foundation
+import OktaIdx
+
+extension OktaIdxAuth.Implementation.Request {
+    class Authenticate: Request<Response>, OktaIdxAuthRemediationRequest {
+        var username: String
```

Comment:
Is it correct that a developer can change `username` or `password` after initialization? Does it make sense to make these values immutable and when a developer wants to execute a new request then new object of `Authenticate` will be created.

Reply:
I'll change these to immutable properties; originally this object was intended to be public to a user, and would be copied before being executed, but I encapsulated it and forgot to change them to `let` values.